src/core/dep-graph.ts

```diff
@@ -327,17 +306,26 @@ class DepGraphImpl implements types.DepGraphInternal {
     return node;
   }
 
-  private pathsFromNodeToRoot(nodeId: string): types.PkgInfo[][] {
+  private pathsFromNodeToRoot(
+    nodeId: string,
+    ancestors: string[] = [],
+  ): types.PkgInfo[][] {
+    if (ancestors.includes(nodeId)) {
+      return [];
```

Comment:
This feels wrong

Reply:
I'll move this condition into another place